### PR TITLE
fix: fetch doesn't respect headers from 1st arg if its type is Request

### PR DESCRIPTION
**Pull request by codercms**
Fixes compatibility with libraries that call the fetch API as `fetch(new Request(...someParams))` without an `init` argument

One of these libraries is https://github.com/sindresorhus/ky (https://github.com/sindresorhus/ky/blob/main/source/utils/timeout.ts)

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/kit/src/runtime/server/page/load_data.js [hunk 1]
```diff
@@ -203,6 +203,15 @@ export function create_universal_fetch(event, state, fetched, csr, resolve_opts)
...
 	return async (input, init) => {
 		const cloned_body = input instanceof Request && input.body ? input.clone().body : null;
+
+		/** @type {HeadersInit | undefined} */
+		let cloned_headers;
+		if (input instanceof Request && input.headers.keys().next().value?.length) {
+			cloned_headers = new Headers(input.headers);
+		} else {
+			cloned_headers = init?.headers;
+		}
```
Comment (ghostdevv) [inline packages/kit/src/runtime/server/page/load_data.js:213]:
```suggestion
		const cloned_headers = input instanceof Request && [...input.headers].length ? new Headers(input.headers) : init?.headers;
```

we could do something like this to match up with `cloned_body` - it's unfortunate there is no size property on headers :/

Comment (codercms) [reply packages/kit/src/runtime/server/page/load_data.js]: @ghostdevv sadly `pnpm lint` fails with this change, and I can't understand why:
```
[warn] src/runtime/server/page/load_data.js
[warn] Code style issues found in the above file. Forgot to run Prettier?
 ELIFECYCLE  Command failed with exit code 1.
```

Is there any way to see an error message? It's my 1st time with prettier

Comment (gtm-nayan) [reply packages/kit/src/runtime/server/page/load_data.js]: The line's probably just too long. You can do a git pull, run format then push again.

**Review by ghostdevv — APPROVED** [review]

**Comment by ghostdevv** [comment]
Should we have a changeset here?

**Comment by codercms** [comment]
@ghostdevv done :)

File: packages/kit/test/apps/basics/test/test.js [hunk 2]
```diff
@@ -264,10 +264,10 @@ test.describe('Load', () => {
 			const payload_b = '{"status":200,"statusText":"","headers":{},"body":"Y"}';
 			// by the time JS has run, hydration will have nuked these scripts
 			const script_contents_a = await page.innerHTML(
-				'script[data-sveltekit-fetched][data-url="/load/serialization-post.json"][data-hash="3t25"]'
+				'script[data-sveltekit-fetched][data-url="/load/serialization-post.json"][data-hash="1vn6nlx"]'
 			);
 			const script_contents_b = await page.innerHTML(
-				'script[data-sveltekit-fetched][data-url="/load/serialization-post.json"][data-hash="3t24"]'
+				'script[data-sveltekit-fetched][data-url="/load/serialization-post.json"][data-hash="1vn6nlw"]'
```
Comment (ghostdevv) [inline packages/kit/test/apps/basics/test/test.js:270]: I missed this initially - is this change deliberate or accidental?

Comment (codercms) [reply packages/kit/test/apps/basics/test/test.js:270]: @ghostdevv this change is deliberate, because the hashes has been changed, now the hashes are the same as in  `POST fetches are serialized` test case , https://github.com/sveltejs/kit/blob/master/packages/kit/test/apps/basics/test/test.js#L243
